### ISSUE TITLE
Latest release does not work in Visual Studio 2015 (Community Edition)

### ISSUE THREAD

**Issue by dthul** [samizzo/nshader#12]
These are the relevant lines from the Log:

```xml
  <entry>
    <record>470</record>
    <time>2017/03/20 09:33:29.121</time>
    <type>Information</type>
    <source>VisualStudio</source>
    <description>Begin package load [NShader]</description>
    <guid>{DD9B8CDE-96A0-4442-BF6F-083A7F43D453}</guid>
  </entry>
  <entry>
    <record>471</record>
    <time>2017/03/20 09:33:29.136</time>
    <type>Error</type>
    <source>VisualStudio</source>
    <description>CreateInstance failed for package [NShader]Source: &apos;mscorlib&apos; Description: Could not load file or assembly &apos;Microsoft.VisualStudio.Shell.15.0, Version=15.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a&apos; or one of its dependencies. The system cannot find the file specified.&#x000D;&#x000A;System.IO.FileNotFoundException: Could not load file or assembly &apos;Microsoft.VisualStudio.Shell.15.0, Version=15.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a&apos; or one of its dependencies. The system cannot find the file specified.&#x000D;&#x000A;File name: &apos;Microsoft.VisualStudio.Shell.15.0, Version=15.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a&apos;&#x000D;&#x000A;   at System.Reflection.RuntimeAssembly.GetType(RuntimeAssembly assembly, String name, Boolean throwOnError, Boolean ignoreCase, ObjectHandleOnStack type)&#x000D;&#x000A;   at System.Reflection.RuntimeAssembly.GetType(String name, Boolean throwOnError, Boolean ignoreCase)&#x000D;&#x000A;   at System.Activator.CreateInstanceFromInternal(String assemblyFile, String typeName, Boolean ignoreCase, BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture, Object[] activationAttributes, Evidence securityInfo)&#x000D;&#x000A;   at System.AppDomain.CreateInstanceFrom(String assemblyFile, String typeName)&#x000D;&#x000A;&#x000D;&#x000A;WRN: Assembly binding logging is turned OFF.&#x000D;&#x000A;To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1.&#x000D;&#x000A;Note: There is some performance penalty associated with assembly bind failure logging.&#x000D;&#x000A;To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].&#x000D;&#x000A;</description>
    <guid>{DD9B8CDE-96A0-4442-BF6F-083A7F43D453}</guid>
    <hr>80004005 - E_FAIL</hr>
    <errorinfo></errorinfo>
  </entry>
```

Version 2.1 works fine though, so I guess something broke when you added VS2017 support.

**Comment by samizzo** [comment]
Ah bugger! Thanks very much, I'll look into that!

**Comment by Xirema** [comment]
I'm having trouble with Visual Studio 2017. Where were these logs pulled from? I'll check my instance and try to figure out if I have a similar error.

**Comment by dthul** [comment]
Visual Studio actually opened a little prompt and pointed me to the location of that log file. I can't check it right now but it was somewhere in my user's AppData folder.

**Comment by dthul** [comment]
Ok, seems to be something like `C:\Users\<username>\AppData\Roaming\Microsoft\VisualStudio\<version>\ActivityLog.xml`

**Comment by samizzo** [comment]
@Xirema What sort of troubles were you having? One thing I noticed with the VS2017 version was that the colour settings didn't appear in Tools -> Options until a shader file was opened. It seems in VS2017 extensions don't initialise until a file type they are registered for is opened.

**Comment by Xirema** [comment]
@samizzo I don't get any error messages, but my .glsl files don't show any coloring. I tried closing and reopening the files, to no change.
![nshader](https://cloud.githubusercontent.com/assets/1447761/24125686/e6798c44-0d97-11e7-9c83-7138d319a6e7.PNG)

**Comment by samizzo** [comment]
Does the extension show up in Help->About in the "Installed products" box?

**Comment by Xirema** [comment]
Okay, I resolved the issue by updating to the *latest* version of VS2017. I was an update behind, and the colors are showing up now.

**Comment by samizzo** [comment]
Oh weird! Ok thanks for letting me know.

**Comment by Ludophonic** [comment]
I just installed NShader in 2017 and had the same problem as Xirema. I got it to work by disabling NShader, restarting Visual Studio, re-enabling NShader and then restarting again.

**Comment by samizzo** [comment]
Thanks for the info @Ludophonic. I created a new issue (issue #13) for the 2017 problem since this report was originally about 2015.

If anyone else has any info about the VS 2017 bug please post it in issue #13.

**Comment by samizzo** [comment]
@dthul I've fixed up the VS2015 issue. NShader was referencing all sorts of VSSDK DLLs from different VS versions, and when I upgrade to VS2017 it referenced the VS2017 versions. So when it loaded on a machine that never had VS2017 installed it would fail.

I've fixed it to now use NuGet to get the VS2013 VSSDK DLLs when building instead of using whatever version of VS you're building in. This way it will use the VS2013 VSSDK interfaces which 2015 and 2017 are also compatible with.

I tested this in an VM so I'm pretty sure everything should be working now. Let me know how it goes. There's a new [release](https://github.com/samizzo/nshader/releases/tag/2.4) available.
